geopandas/array.py

```diff
@@ -289,6 +289,34 @@ def sindex(self):
             self._sindex = get_sindex_class()(self.data)
         return self._sindex
 
+    def sindex_generated(self):
+        """Retruns `True` if the spatial index (`.sindex` property)
+        has been generated and `False` otherwise.
+        Note that spatial indexes may lazily generate the index,
```

Comment:
I felt that it was important to clarify because if a user is using this, they are probably wanting to check which geometry iterable they want to use as the spatial index and which as the query in order to improve performance. Since in the future the default sindex implementation will be lazily built, I feel that users could get very confused as to why it takes a long time the first time they query even if the index was supposed to be generated already.

That said, your text is much better, so happy to replace it.